### PR TITLE
Feature: add region explicitly in aws provider to avoid regressions

### PR DESCRIPTION
**Pull request by andylamp**
## Description

This addresses the regression observed in certain configurations as described in issue #14. 

The PR refactors the project to take the `region` from variables but only _if and only if provided explicitly_. If not, the current behavior is maintained. This refactor means that all modules within the project take the provider region as a parameter which is used for all non aliased (_i.e._ non-global) ones.

## Context

Resolves regressions observed in #14.

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Comment by andylamp** [comment]
is any maintainer around?

**Comment by Tomdango** [comment]
Hi @andylamp,

Sorry for the slowness in getting this resolved. There have been other priorities which means this project has slightly slipped - we'll be looking for more maintainers of the project soon.

Could you recreate the branch without the `aws_wafv2_web_acl_logging_configuration` commented out? The issue with this is being fixed under #17.

**Comment by andylamp** [comment]
hey @Tomdango! thanks for reaching out and sorry for pestering about this but we needed to deploy the module. Due the issues it had we could not.

Firstly, at first apply - at least with the latest providers and terraform versions - the  resource `aws_wafv2_web_acl_logging_configuration` has a `count` which cannot be inferred at creation and causes issues. It requires to use `target` to selectively apply them. 

Commenting that was the easiest way to get this running as it was not of high priority to us to have this going, but feel this is a significant issue.

Another is that the `concurrency` for the lambdas is hardcoded and there is no parameter to tweak it, so I made a change to account for that. Please see the branch [here](https://github.com/andylamp/terraform-aws-opennext/tree/v1.0.5) in my own fork.

Note: these again used the philosophy of enhance functionality but not introduce breaking changes, so every change is a drop-in replacement without affecting current functionality.

LMK what you think.

**Comment by andylamp** [comment]
@Tomdango, just asking again about this for the last time... otherwise, I think I'll just fork and publish my own version in TF repositories...

**Comment by andylamp** [comment]
Closing this due to the reasons state above.

File: modules/opennext-cloudfront/waf.tf [hunk 1]
```diff
@@ -121,78 +121,78 @@ resource "aws_wafv2_web_acl" "cloudfront_waf" {
...
+#      }
+#
+#      dynamic "uri_path" {
+#        for_each = redacted_fields.value.uri_path == true ? [true] : []
+#
+#        content {}
+#      }
+#    }
+#  }
+#}
```
Comment (Tomdango) [inline modules/opennext-cloudfront/waf.tf:198]: These can now be uncommented.

**Review by Tomdango — CHANGES_REQUESTED** [review]
The changes look good - just the commented sections need reintroducing and then I'll merge and create a new release.

**Comment by andylamp** [comment]
@Tomdango thanks a lot for the visibility, sorry was not aware this was not actively maintained... I know the burden open source can bring... Just wanted to have these changes since we are actively using this library in our FE pipelines. BTW, given that we are actively using this perhaps I could chip in to help you maintain this? If that's OK with you as well...

**Comment by andylamp** [comment]
ah, also - the latest version is [v1.0.6](https://github.com/andylamp/terraform-aws-opennext/tree/v1.0.6) which includes a few other fixes as well... I'll merge these changes an open a new PR regarding this.

**Comment by Tomdango** [comment]
@andylamp Feel free to send an email over to thomas.judd-cooper1@nhs.net and we'll take it from there 🙂

**Pull request merged**